packages/redux-dynamic-modules-react/src/DynamicModuleHook.ts

```diff
@@ -0,0 +1,21 @@
+import { useContext, useEffect } from "react";
+import { IModuleStore, IModule } from "redux-dynamic-modules";
+//@ts-ignore
+import { Provider, ReactReduxContext } from "react-redux";
+
+export function useModules(modules: IModule<any>[]) {
```

Comment:
So I have been thinking about this, seems like this will break rules of react. But I could be wrong so please correct me.
In HOC solution, we were adding the module in componentWillMount, so adding the module to store before render.

But here due to line 19, we are causing side effects including running initial sagas during render, which seems not correct to me.

Additionally we can not use useEffects as well, as per the useEffects [doc](https://reactjs.org/docs/hooks-effect.html) "If you’re familiar with React class lifecycle methods, you can think of useEffect Hook as **componentDidMount, componentDidUpdate, and componentWillUnmount combined."**.

What are your thoughts?

Reply:
We were never adding in `willMount`, we were adding in the constructor, which is apparently materially different. 

Mohit is also correct, that all of this needs to go into useEffects so that it is not called with every render of this component.

I think we need to figure out a way to do all of this work in the `did*` lifecycle methods. We run into the same problem on #43 where the pre-render lifecycle methods are now deprecated

Reply:
Yes ... I want to see how react-redux switches to hooks, May be we will get some inspiration or ideas.

Reply:
We can not use useEffect as it happens AFTER the component is rendered. Or am I missing anything?